File: frontend/src/types/prediction.ts
export interface PredictionRequest {
  BHK: number;
  Floor: number;
  Bathroom: number;
  Balcony: number;
  Covered_Parking: number;
  Open_Parking: number;
  Garden_Park: number;
  Main_Road: number;
  Pool: number;
  Location: string;
  Transaction: string;
  Furnishing: string;
}

export interface PredictionResponse {
  predicted_price: number;
}

File: backend/app/schemas/prediction.py
from pydantic import BaseModel

class PredictionRequest(BaseModel):
    BHK: int
    Floor: int
    Bathroom: float
    Balcony: float
    Covered_Parking: int
    Open_Parking: int
    Garden_Park: int
    Main_Road: int
    Pool: int
    Location: str
    Transaction: str
    Furnishing: str

class PredictionResponse(BaseModel):
    predicted_price: float